onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -99,34 +114,16 @@ def run(self):
                 utils.publish_data_dictionary(self.pub_pos, utils.get_axes(), self.pose)
                 self.pose = None
 
-            elif self.powers:
-                if self.last_powers is None or self.powers != self.last_powers:
-                    self.enable_loops()
-                    # Hold Position on the current state
-                    self.hold = dict(self.state)
-                    self.last_powers = dict(self.powers)
-
-                # Nonzero entries bypass PID
-                # If any nonzero xy power, disable those position pid loops
-                if self.powers['x'] != 0 or self.powers['y'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['x', 'y'], False)
-
-                # If any nonzero roll or pitch power, disable those position pid loops
-                if self.powers['roll'] != 0 or self.powers['pitch'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['roll', 'pitch'], False)
-
-                # If nonzero yaw power, disable those position pid loops
-                if self.powers['yaw'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['yaw'], False)
-
-                # If any nonzero z power, disable those position pid loops
-                if self.powers['z'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['z'], False)
-
-                utils.publish_data_dictionary(self.pub_power, utils.get_axes(), self.powers)
-                # Publish current state to the desired state for PID
-                utils.publish_data_dictionary(self.pub_pos, utils.get_axes(), self.hold)
-                self.powers = None
+            elif self.twist:
+                self.enable_loops()
+                self.disable_pos_loop()
```

Comment:
I disagree. The solution you propose has same issue of queue size limits, as we would first have to disable both loops with a self.disable_loops() call then only enable the vel loop. This can only be avoided we have a disable/enable function for both pos and vel.